src/main/java/anodot/stage/destination/AnodotTarget.java

```diff
@@ -139,6 +147,40 @@ public void write(Batch batch) throws StageException {
         }
     }
 
+    private Record getLastRecord(Batch batch) {
+        Iterator<Record> iter = batch.getRecords();
+        Record record = null;
+        while (iter.hasNext()) {
+            record = iter.next();
+        }
+        return record;
+    }
+
+    private void sendOffsetToAgent(String offset, String pipelineId) throws Exception {
+        URLConnection con = new URL(conf.agentOffsetUrl + pipelineId).openConnection();
```

Comment:
I there any simpler method to make post request using `streamsets-datacollector-httpcommonlib` ?

Reply:
Also, i think we need to set some kind of timeout in case agent is not responsible

Reply:
do we also need to close `HttpURLConnection`?